docs/src/main/sphinx/functions/json.rst

```diff
@@ -2,6 +2,1174 @@
+
+.. code-block:: text
+
+    <path>.*
+
+Let ``<path>`` return a sequence of three JSON objects:
+
+.. code-block:: text
+
+    {"customer" : 100, "region" : "AFRICA"},
+    {"region" : "ASIA"},
+    {"customer" : 300, "region" : "AFRICA", "comment" : null}
+
+The results is:
+
+.. code-block:: text
+
+    100, "AFRICA", "ASIA", 300, "AFRICA", null
+
+It is required that all items in the input sequence are JSON objects.
+
+The order of the returned values follows the order of the original JSON
+objects. However, within objects, the order is arbitrary.
+
+array accessor
+''''''''''''''
+
+Returns the elements at the specified indexes for each JSON array in the input
+sequence. Indexes are zero-based.
+
+.. code-block:: text
+
+    <path>[ <subscripts> ]
+
+The ``<subscripts>`` list contains one or more subscripts. Each subscript
+specifies a single index or a range (ends inclusive):
+
+.. code-block:: text
+
+    <path>[<path1>, <path2> to <path3>, <path4>,...]
+
+After the input sequence is evaluated, in the lax mode, all non-array items
+in the sequence are wrapped in singleton arrays. Array wrapping replaces the
+usual unwrapping, because the array accessor requires that all the input items
+are JSON arrays.
+
+For every array in the input sequence, all subscript indexes are computed, with
+the variable ``last`` referring to the last index of the current array. For a
+singleton subscript ``<path1>``, the result must be a singleton numeric item.
+For a range subscript ``<path2> to <path3>``, two numeric items are expected.
+The expected array elements are added in order to the output sequence.
+
+Let ``<path>`` return a sequence of three JSON arrays:
+
+.. code-block:: text
+
+    [0, 1, 2], ["a", "b", "c", "d"], [null, null]
+
+The following expression returns a sequence containing the last element from
+every array:
+
+.. code-block:: text
+
+    <path>[last] --> 2, "d", null
+
+The following expression returns the third and fourth element from every array:
+
+.. code-block:: text
+
+    <path>[2 to 3] --> 2, "c", "d"
+
+Note that the first array does not have the fourth element, and the last array
+does not have the third or fourth element. Accessing non-existent elements is a
+structural error. In the strict mode, it causes the path expression to fail. In
+the lax mode, such errors are suppressed, and only the existing elements are
+returned.
+
+Incorrect range is another example of a structural error: ``5 to 3``.
+
+Note that the subscripts may overlap, and they do not need to follow the
+element order. The order in the returned sequence follows the subscripts:
+
+.. code-block:: text
+
+    <path>[1, 0, 0] --> 1, 0, 0, "b", "a", "a", null, null, null
+
+wildcard array accessor
+'''''''''''''''''''''''
+
+Returns all elements of each JSON array in the input sequence.
+
+.. code-block:: text
+
+    <path>[*]
+
+After the input sequence is evaluated, in the lax mode, all non-array items
```

Comment:
"In lax mode, any non-array items resulting from the evaluation of the input sequence are wrapped into single-element arrays."